modules/engines/engine-mail-destination/src/main/java/com/sap/xsk/mail/DestMailConfigProvider.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.mail;
+
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationAccessException;
+import com.sap.xsk.api.destination.CloudPlatformDestinationFacade;
+import org.eclipse.dirigible.api.v3.mail.api.IMailConfigurationProvider;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.Properties;
+
+public class DestMailConfigProvider implements IMailConfigurationProvider {
+  private static final String MAIL_USER = "mail.user";
+  private static final String MAIL_PASSWORD = "mail.password";
+  private static final String MAIL_TRANSPORT_PROTOCOL = "mail.transport.protocol";
+  private static final String MAIL_SMTPS_HOST = "mail.smtps.host";
+  private static final String MAIL_SMTPS_PORT = "mail.smtps.port";
+  private static final String MAIL_SMTPS_AUTH = "mail.smtps.auth";
+  private static final String MAIL_SMTP_HOST = "mail.smtp.host";
+  private static final String MAIL_SMTP_PORT = "mail.smtp.port";
+  private static final String MAIL_SMTP_AUTH = "mail.smtp.auth";
+  private static List<String> MAIL_PROPERTIES = List.of(MAIL_USER, MAIL_PASSWORD, MAIL_TRANSPORT_PROTOCOL, MAIL_SMTPS_HOST, MAIL_SMTPS_PORT,
+      MAIL_SMTPS_AUTH, MAIL_SMTP_HOST, MAIL_SMTP_PORT, MAIL_SMTP_AUTH);
+
+  private static final String PROVIDER_NAME = "destination";
+  private static final String DESTINATION_NAME = "Mail";
```

Comment:
this should be configured from the environment like `MAIL_SERVER_DESTINATION_NAME`

Reply:
Fixed in a8928233f6d9d4d2935b4c96a230cd2535e46687